src/atomate2/openmm/jobs/base.py

```diff
@@ -0,0 +1,156 @@
+from typing import Optional, Union, Callable
+from jobflow import job, Maker
+from dataclasses import dataclass
+from pathlib import Path
+from pymatgen.io.openmm.sets import OpenMMSet
+from src.atomate2.openmm.schemas.openmm_task_document import OpenMMTaskDocument
+from openmm import (
+    Platform,
+    Context,
+)
+from typing import (
+    Union,
+    Optional,
+    Dict,
+)
+from openmm.app import (
+    DCDReporter,
+    StateDataReporter,
+    PDBReporter,
+)
+from openmm.openmm import (
+    MonteCarloBarostat,
+    AndersenThermostat,
+)
+from openmm.unit import (
+    kelvin,
+    atmosphere,
+    pico,
+    second
+)
+import os
+
+
+def openmm_job(method: Callable):
+    """
+
+    Parameters
+    ----------
+    method : callable
+        A BaseOpenmmMaker.make method. This should not be specified directly and is
+        implied by the decorator.
+
+    Returns
+    -------
+    callable
+        A decorated version of the make function that will generate OpenMM jobs.
+    """
+    return job(method, output_schema=OpenMMTaskDocument)
+
+
+@dataclass
+class BaseOpenmmMaker(Maker):
+    """
+    todo: doc string
+    """
+    name: str = "base openmm job"
+    get_simulation_kwargs: Optional[dict] = None
+    dcd_reporter_kwargs: Optional[dict] = None
+    pdb_reporter_kwargs: Optional[dict] = None
+    state_reporter_kwargs: Optional[dict] = None
```

Comment:
Maybe we should only expose specific kwargs for each reporter? For example, state has like 15 kwargs but I don't think we want to expose them all. In fact, I don't know that we want to expose any other than reportInterval. I also don't think we want the user to have to reference the OpenMM docs to use this package. How could we address that?

Reply:
I'm in favor of exposing all kwargs to provide full control, but I agree we don't want the user to have to reference documentation in order to add a reporter. What if in addition to the optional kwargs there was an optional boolean argument to turn on/off reporters? When a reporter is set to True the kwargs are passed to the reporter init. If kwargs weren't specified defaults would be used.